packages/nextjs/hooks/scaffold-eth/useScaffoldEventHistory.ts

```diff
@@ -0,0 +1,122 @@
+import { ContractAbi, ContractName } from "~~/utils/scaffold-eth/contract";
+import { ExtractAbiEventNames } from "abitype";
+import { useProvider, useContract } from "wagmi";
+import { useDeployedContractInfo } from "~~/hooks/scaffold-eth";
+import { ethers } from "ethers";
+import { useEffect, useState } from 'react';
+
+/**
+ * @dev reads events from a deployed contract
+ * @param config - The config settings
+ * @param config.contractName - deployed contract name
+ * @param config.eventName - name of the event to listen for
+ * @param config.fromBlock - the block number to start reading events from
+ * @param config.filters - filters to be applied to the event (parameterName: value)
+ * @param config.blockData - if set to true it will return the block data for each event (default: false)
+ * @param config.transactionData - if set to true it will return the transaction data for each event (default: false)
+ * @param config.receiptData - if set to true it will return the receipt data for each event (default: false)
+ */
+export const useScaffoldEventHistory = <
+  TContractName extends ContractName,
+  TEventName extends ExtractAbiEventNames<ContractAbi<TContractName>>,
+>({
+  contractName,
+  eventName,
+  fromBlock,
+  filters,
+  blockData,
+  transactionData,
+  receiptData,
+}: {
+  contractName: TContractName;
+  eventName: TEventName;
+  fromBlock: number;
+  filters?: any;
+  blockData?: boolean;
+  transactionData?: boolean;
+  receiptData?: boolean;
+}) => {
+  const [events, setEvents] = useState(<any>[]);
+  const [isLoading, setIsLoading] = useState(true);
+  const { data: deployedContractData, isLoading: deployedContractLoading } = useDeployedContractInfo(contractName);
+  const provider = useProvider();
+
+  const contract = useContract({
+    address: deployedContractData?.address,
+    abi: deployedContractData?.abi,
+    signerOrProvider: provider,
+  });
+
+  useEffect(() => {
+    async function readEvents() {
+      if (!contract) {
+        console.log("contract not found");
+        return;
+      }
```

Comment:
I think we should probably set `isLoading` to false here before returning. This condition will only be met if we finished loading the `deployedContractData` and this data doesn't result in a valid contract. All that we were waiting for is settled here. Maybe we can add an `error` state that we set and return from the hook, rather than the console.log.

Reply:
Yup the `error` state will be really great, also I think instead we should use `console.error` cause thats what are preferring while logging error 🙌

Reply:
Added the error state, set isLoading to false in error cases, and changed to use console.error.

I had to change the useDeployedContractInfo hook, to set the isLoading flag to false if the contract code is not found. https://github.com/scaffold-eth/se-2/pull/272/commits/fa6099cb5d5e07b2ccb32127c712cb8aba1dcc7f